File: domain/src/main/kotlin/com/readler/domain/usecase/OpenBookUseCase.kt
package com.readler.domain.usecase

import com.readler.core.model.OpenedBook
import com.readler.domain.repository.BookRepository

class OpenBookUseCase(
    private val bookRepository: BookRepository
) {
    suspend operator fun invoke(bookId: String): OpenedBook = bookRepository.openBook(bookId)
}

File: lib/domain/usecase/open_book_use_case.dart
import '../../core/model/opened_book.dart';
import '../repository/book_repository.dart';

class OpenBookUseCase {
  final BookRepository _bookRepository;

  OpenBookUseCase(this._bookRepository);

  Future<OpenedBook> call(String bookId) => _bookRepository.openBook(bookId);
}
